# Python_scripts

Calculate_N50_gc_N_of_genome.py


```
usage: python3 Calculate_N50_gc_N_of_genome.py -i <genome.fa/genome.fa.gz> -o <output.txt> -m <N50|len>

This script is used to calculate the scaffold length, N50, GC% and N% of the genome.

required arguments:
  -i FASTA FILE, --in FASTA FILE
                        Input sequence file in FASTA format
  -o OUTPUT, --out OUTPUT
                        Output file will be labelled with this name. WARNING: do not provide a path
  -m MODE, --mode MODE  Specify which analysis mode to run.
                        There are two valid modes:
                        -m len for calculating the length, GC%, and N% of each scaffold or genome
                        -m N50 for calculating the scaffold N50 of the genome

optional arguments:
  -h, --help            Show this help message and exit
  ```


Concatenating_Single_Copy_Orthologue.py

```
usage: python3 Concatenating_Single_Copy_Orthologue.py -i <Single_Copy_Orthologue_Sequences_folder_of_OrthoFinder> -a <MultipleSequenceAlignments_folder_of_OrthoFinder> -o <output.phy>

This script is used to concatenate the single copy orthologue generated by OrthoFinder to the phy format for the phylogenomic analysis

required arguments:
  -i IN, --in IN        Single_Copy_Orthologue_Sequences folder of OrthoFinder
  -a ALIGN, --align ALIGN
                        MultipleSequenceAlignments folder of OrthoFinder
  -o OUTPUT, --out OUTPUT
                        Output file in phylip format. WARNING: do not provide a path

optional arguments:
  -h, --help            Show this help message and exit
  ```
